ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/StorageExceptionManager.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth.sync;
+
+import org.hyperledger.besu.plugin.services.exception.StorageException;
+
+public final class StorageExceptionManager {
+
+  private static final String rocksdbClassName = "RocksDBException";
+  private static final String ERR_BUSY = "Busy";
+  private static final String ERR_LOCK_TIMED_OUT = "TimedOut(LockTimeout)";
+
+  public static boolean canRetryOnError(final StorageException e) {
+    return e.getMessage().contains(rocksdbClassName)
+        && (e.getMessage().contains(ERR_BUSY) || e.getMessage().contains(ERR_LOCK_TIMED_OUT));
+  }
```

Comment:
less brittle to use enum values

```suggestion
      return Optional.of(e.getCause())
          .filter(z -> z instanceof RocksDBException)
          .map(RocksDBException.class::cast)
          .map(RocksDBException::getStatus)
          .map(Status::getCode)
          .map(RETRYABLE_STATUS_CODES::contains)
          .orElse(false);
```

Reply:
cannot work because the exception is like that
“Unexpected exception in pipeline. Aborting.“,”throwable”:” org.hyperledger.besu.plugin.services.exception.StorageException: org.rocksdb.RocksDBException: Busy"

the cause will be a Storage exception and the message "org.rocksdb.RocksDBException: Busy"